FreeRDP/FreeRDP
# [core,arm] Waiting until an AVD instance is up

## Preparations
* [x] The branch is re-based.
* [x] The commit history is cleaned up.
* [x] Formatted using clang-format
* [x] `scan-build` was run.
  * `master` branch - 155 bugs found.
  * `avd-auto-reconnect-poc` branch - (built `master` and the the current branch) 0 bugs found.
* [ ] Run a compiler at pedantic level.

## Accepting the contribution
* [x] Give a proper name.
* [x] Add a brief description.
* [x] Give instructions on how to test the changes.

## Description

#9214 

In addition to the issue I would like to:
* prepare correct logging / function exits for cases where the response isn't JSON or there is no the `Code` property.
* move the functionality to a separate functions (maybe even outside `arm.c`).
* update the "running" message to match the Web-client version.
* no memory leaks on successful pass, but other ways are needed to be verified.

It would be ideal to add such functionality for the next case when we lost our internet connection or connected to a VPN:

```
[transport_default_write]: ERRCONNECT_CONNECT_TRANSPORT_FAILED [0x0002000D]
[transport_check_fds]: ERRCONNECT_CONNECT_TRANSPORT_FAILED [0x0002000D]
[transport_check_fds]: TODO: Trying to set error code ERRCONNECT_CONNECT_TRANSPORT_FAILED, but ERRCONNECT_CONNECT_TRANSPORT_FAILED already set!
[input_ensure_client_running]: [APPLICATION BUG] input functions called after the session terminated
```

## Instructions on how to test

1. Disable an AVD instance using Azure Portal or the `shudown /s` command in Command Prompt.
2. Connect to the instance.
3. The following messages appears:
```
Browse to: https://login.microsoftonline.com/common/oauth2/v2.0/authorize?client_id=a85cf173-4192-42f8-81fa-777a763e6e2c&response_type=code&scope=https%3A%2F%2Fwww.wvd.microsoft.com%2F.default&redirect_uri=https%3A%2F%2Flogin.microsoftonline.com%2Fcommon%2Foauth2%2Fnativeclient
Paste redirect URL here:
<redirect-URL>
[10:10:41:501] [49251:0000c064] [INFO][com.freerdp.core.gateway.arm] - [arm_resolve_endpoint]: Please wait... VM is starting running.
[10:11:42:259] [49251:0000c064] [INFO][com.freerdp.core.gateway.arm] - [arm_resolve_endpoint]: Please wait... VM is starting running.
```
4. After a few minutes, FreeRDP will connect to the instance.
